### PR TITLE
Bump to version 0.1.0 and prepare for release

### PR DESCRIPTION
**Pull request by fritzo**
Blocked by https://github.com/pyro-ppl/pyro/pull/2075

I plan to upload to PyPI from this branch, then tag and merge. Here is a preview of the [pypi page](https://test.pypi.org/project/funsor/0.0.4/).

Many *.py files are touched because I ran `sed -i 's/contextlib2/contextlib/g'` then `make format` which runs isort.

## Tasks
- [x] code review
- [ ] upload to PyPI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@ install:
     # Keep track of Pyro dev branch
     - pip install https://github.com/pyro-ppl/pyro/archive/dev.zip
 
+    # Keep track of pyro-api master branch
+    - pip install https://github.com/pyro-ppl/pyro-api/archive/master.zip
```
Comment (fritzo) [inline .travis.yml]: @neerajprad I've added this so we can use numeric versioning in setup.py, which is required by PyPI

**Comment by fritzo** [comment]
@jpchen is there anything special we need to do for 0.1.0 to show up on readthedocs as a stable version?

**Comment by jpchen** [comment]
> is there anything special we need to do for 0.1.0 to show up on readthedocs as a stable version

not afaik

File: setup.py [hunk 2]
```diff
@@ -2,18 +2,17 @@
 
 setup(
     name='funsor',
-    version='0.0.0',
+    version='0.1.0',
```
Comment (neerajprad) [inline setup.py]: Would you like to populate the long description for pypi? Its totally cool if you are looking to skip it - just wanted to make sure.

Comment (fritzo) [reply setup.py]: Good point, I'll add that.

Comment (fritzo) [reply setup.py]: done.

File: setup.py [hunk 3]
```diff
@@ -1,19 +1,36 @@
+import sys
+
 from setuptools import find_packages, setup
 
+# READ README.md for long description on PyPi.
+# This requires uploading via twine, e.g.:
+# $ python setup.py sdist bdist_wheel
+# $ twine upload --repository-url https://test.pypi.org/legacy/ dist/*  # test version
```
Comment (neerajprad) [inline setup.py]: Nice to have this comment right here!

**Comment by neerajprad** [comment]
Feel free to merge.

**Comment by fritzo** [comment]
I'll merge after successfully uploading to PyPI, I just don't want to merge then realize I botched the setup.py.

**Pull request merged**